components/gmail_custom_oauth/sources/new-labeled-email/new-labeled-email.mjs

```diff
@@ -0,0 +1,116 @@
+import { DEFAULT_POLLING_SOURCE_TIMER_INTERVAL } from "@pipedream/platform";
+import gmail from "../../gmail_custom_oauth.app.mjs";
+
+export default {
+  gmail,
+  key: "gmail_custom_oauth-new-labeled-email",
+  name: "New Labeled Email",
+  description: "Emit new event when a new email is labeled.",
+  type: "source",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    gmail,
+    db: "$.service.db",
+    timer: {
+      label: "Polling interval",
+      description: "Pipedream will poll the Monday API on this schedule",
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: DEFAULT_POLLING_SOURCE_TIMER_INTERVAL,
+      },
+    },
+    label: {
+      propDefinition: [
+        gmail,
+        "label",
+      ],
+    },
+  },
+  methods: {
+    ...gmail.methods,
+    getMessage({
+      messageId, ...opts
+    }) {
+      return this.gmail._makeRequest({
+        path: `messages/${messageId}`,
+        ...opts,
+      });
+    },
+    listMessages(opts = {}) {
+      return this.gmail._makeRequest({
+        path: "messages",
+        ...opts,
+      });
+    },
```

Comment:
Methods `getMessage` and `listMessages` are already included in the app file. They're located in `gmail.app.mjs` which is included as the base for `gmail_custom_oauth.app.mjs`. We should use the existing methods instead of adding new ones.

Reply:
@luancazarine Are you still working on this part?